docs/semgrep-app/notifications.md

```diff
@@ -277,31 +277,147 @@ Automated comments on GitLab merge requests are displayed as follows:
 
 :::note
-GitLab MR comments are only available to logged-in Semgrep users, as they require a Semgrep API token.
+GitLab MR comments are only available to logged-in Semgrep Cloud Platform users.
+:::
+
+### Bitbucket pull request comments
+
+To configure Semgrep PR comments in your Bitbucket PRs, follow the steps described in the subsections below.
+
+#### Creating a personal access token
+
+Create a personal access token to authenticate to the Bitbucket API. There are two ways in which you can create personal access tokens depending on the Bitbucket plan you use:
+
+- **Workspace access token**: If you use the Bitbucket Cloud Premium plan, you can create a workspace access token. This option saves time because you can create one access token for all repositories in the workspace. See [Creating a workspace access token](/semgrep-app/notifications/#creating-and-adding-a-workspace-access-token).
+- **Repository access token**: If you are **not** using the Bitbucket Cloud Premium plan, you have to create a separate repository access token for each repository where you want to use Semgrep. See [Creating a repository access token](/semgrep-app/notifications/#creating-and-adding-a-repository-access-token).
+
+##### Creating and adding a workspace access token
+
+:::info Prerequisite
+Use the procedure described in this section if you use the **Bitbucket Cloud Premium** plan. If you are **not** using the Bitbucket Cloud Premium plan, you have to create a separate repository access token for each repository where you want to use Semgrep. See [Creating a repository access token](/semgrep-ci/running-semgrep-ci-with-semgrep-app/#creating-and-adding-a-repository-access-token).
+:::
+
+Create a workspace access token in Bitbucket (only available if you have a Bitbucket Cloud Premium plan). Follow the instructions in [Create a Workspace Access Token](https://support.atlassian.com/bitbucket-cloud/docs/create-a-workspace-access-token/). Alternatively, follow these steps to create a workspace access token:
+
+1. In the Bitbucket workspace, click the cogwheel icon next to your profile icon in the top right.
+1. Click **Workspace settings** > **Access tokens**.
+1. Click **Create Workspace Access Token**.
+1. Create a token name. Note: As Semgrep comments to your PRs, use **Semgrep** as the name for this token.
+1. Under **Scopes**, select the **Read**, and **Write** permissions for pull requests.
+1. Click **Create**.
+1. Copy the value of the workspace access token, and then click **Close**.
+1. Click **Workspace variables**.
+1. Create a new workspace variable by entering **PAT** into the **Name** field, and then paste the workspace access token in the **Value** field.
+1. Enable the **Secured** option, and then click **Add**.
+
+Create and add a `SEMGREP_APP_TOKEN` to establish the communication between your workspace and the Semgrep Cloud Platform by following these steps:
```

Comment:
Suggest to remove lines 357 to 363 and to do the same for the subsequent section.

Reply:
@s-santillan do you mean to delete Creating and adding a repository access token? But users **without** Bitbucket Cloud Premium cannot do otherwise. The only way they can get our PR comments is by adding a repository access token (contrary to Bitbucket Cloud Premium users who can add a workspace access token).

Why do want to delete this section? My main concern is - the goal should be that users successfully get Semgrep messages to their PRs. Do you have an alternative here?

Reply:
Or do you mean the part about the `SEMGREP_APP_TOKEN` in the subsequent section?

Each part of the docs should be as independent as possible. In some sections, you can imagine that our users went just to this page to see how to enable the comments. Their sole goal is to have PR or MR comments, nothing more. Maybe we can make the `SEMGREP_APP_TOKEN` part optional? Also, how to add the token is different in every CMS... so documenting this seems to make sense.

Reply:
By the time the users already get to adding comments, they will have already set up the SEMGREP_APP_TOKEN. 
Flow is like this:
1. Users add repo through Semgrep Cloud PlatformP -> THIS step adds the SEMGREP_APP_TOKEN. There is no way for a user to scan a repo with Semgrep Cloud Platform without it. **This is true for all plans, in all SCMs**, whether it is GitHub's free or paid tier or Bitbucket's free or paid tier. It establishes the connection between the repo and SCP. 
2. Users enable various notifications (PR/MR comments, Slack, etc.)

I think the reason why SEMGREP_APP_TOKEN was not "clear" is because onboarding for a BitBucket repo hasn't been added in the docs. You can however see the SEMGREP_APP_TOKEN being added in various other guides.

Reply:
Here's the guide for setting up a connection in CI: https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-with-semgrep-app/#ci-providers-listed-within-semgrep-cloud-platform-such-as-github-actions-gitlab-cicd-jenkins

Reply:
I'll answer this first:

> Note: I am also not sure that a header **CI providers listed within Semgrep Cloud Platform (such as GitHub Actions, GitLab CI/CD, Jenkins)** is easily understood as the place to search for: Creating SEMGREP_APP_TOKEN.

The parent section is [Setting up the CI job and Semgrep Cloud Platform Connection](https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-with-semgrep-app/#setting-up-the-ci-job-and-semgrep-cloud-platform-connection). It subsections are "CI providers listed within Semgrep Cloud Platform" and "Other CI providers" because the steps differ between those two categories.

Does that answer your question?

Reply:
The admonition is great, putting thumbs up on the last two comments. My main question is: How to actually copy the token from the Cloud Platform so users do not need to create a new one every time? @s-santillan

Reply:
Mentioned above: @s-santillan ![Screenshot 2023-04-03 at 16 41 24](https://user-images.githubusercontent.com/26669307/229543689-feb10a7c-6ecd-4e94-9edb-1db94a1145b1.png)

Reply:
By design, you (admin, member, etc.) can't copy a token after it's been created. Just make a new token if you need to. "If you already have a token configured" refers to the act of copying the token into the CI provider as a secret, which configures the connection.